app/models.py

```diff
@@ -2364,3 +2384,18 @@ def serialize(self):
             "updated_at": get_dt_string_or_none(self.updated_at),
             "logged_in_at": get_dt_string_or_none(self.logged_in_at),
         }
+
+
+class LetterAttachment(db.Model):
+    __tablename__ = "letter_attachment"
+
+    id = db.Column(UUID(as_uuid=True), primary_key=True, nullable=False, default=uuid.uuid4)
+
+    created_at = db.Column(db.DateTime, nullable=True, default=datetime.datetime.utcnow)
+    created_by_id = db.Column(UUID(as_uuid=True), db.ForeignKey("users.id"), nullable=True)
+    updated_at = db.Column(db.DateTime, nullable=True, onupdate=datetime.datetime.utcnow)
+    updated_by_id = db.Column(UUID(as_uuid=True), db.ForeignKey("users.id"), nullable=True)
+
+    service_id = db.Column(UUID(as_uuid=True), db.ForeignKey("services.id"), primary_key=True)
```

Comment:
Would be good to know why we went with service_id vs template_id here.

Reply:
So, `service_id` is definitely something we could drop.

I've excluded `template_id` because we've already got that relationship defined from Template.attachment_id so having the reverse isn't super useful.

The main reason I've added this is because it vastly simplifies us finding out which service the attachment was uploaded by. If the attachment is removed from a template, we make a new template version and then the attachment won't be related to any records in the `template` table - only `template_history`. It's a bit more mentally painful to then have to go attachment->template_history->service to answer that simple (and fairly reasonable I think) question of which service 'owns' the file.

Reply:
I don't think it would be that difficult to add if we need it later. It would be a matter of 1 SQL in the migration. And we definitely don't need it now. I think it would be useful if another dev, like @leohemsted , could weigh in.

I think the uses we need to think about for currently planned functions are:
- viewing the attachment on template page
- what happens when users deletes or replaces the attachment
- how easy it will be to make a rule to delete old attachments from S3. If we have an "archived" column, then when it is `archived`, `updated_at` is updated. We could also just have `archived_at` instead. Then it's easy to check which attachments have been archived a month or more ago and delete them from S3.
Without that the check becomes more awkward.
- how easy it is to re-assemble the letter in case of an issue with template preview, which happened in the past. So if attachment for the letter changes, how easy can we find the right attachment for when notification was sent and stitch it all together again.

Reply:
I have renamed to archived_at and archived_by_id :)